### PR TITLE
Params: python-like interface

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/ui/qt/widgets/controls.hpp [hunk 1]
```diff
@@ -109,12 +109,12 @@ class ParamControl : public ToggleControl {
...
   ParamControl(const QString &param, const QString &title, const QString &desc, const QString &icon, QWidget *parent = nullptr) : ToggleControl(title, desc, icon, parent) {
     // set initial state from param
-    if (Params().read_db_bool(param.toStdString().c_str())) {
+    if (Params().getBool(param.toStdString().c_str())) {
       toggle.togglePosition();
     }
     QObject::connect(this, &ToggleControl::toggleFlipped, [=](int state) {
       char value = state ? '1' : '0';
-      Params().write_db_value(param.toStdString().c_str(), &value, 1);
+      Params().put(param.toStdString().c_str(), &value, 1);
```
Comment (pd0wm) [inline selfdrive/ui/qt/widgets/controls.hpp:117]: Maybe we should make a `putBool` helper as well.

File: selfdrive/common/params.h [hunk 2]
```diff
@@ -12,35 +13,40 @@ class Params {
...
+
+  template <class T>
+  std::optional<T> get(const char *key, bool block = false) {
+    std::istringstream iss(get(key, block));
+    T value{};
+    iss >> value;
+    return iss.fail() ? std::nullopt : std::optional(value);
+  }
+
+  inline bool getBool(const char *key) {
```
Comment (pd0wm) [inline selfdrive/common/params.h:42]: Interesting that only `1` becomes true, not any other values. Did you find documentation for that? I couldn't find anything.

Seems that `1\n` also evaluates to true, but that's fine with me.

Comment (deanlee) [reply selfdrive/common/params.h:42]: fixed.

Comment (deanlee) [reply selfdrive/common/params.h:42]: fixed

Comment (pd0wm) [reply selfdrive/common/params.h:42]: Yeah, maybe good to keep old behavior

**Pull request merged**